datafusion/physical-plan/src/lib.rs

```diff
@@ -91,16 +91,26 @@ pub use datafusion_physical_expr::{
 pub use crate::stream::EmptyRecordBatchStream;
 pub use datafusion_execution::{RecordBatchStream, SendableRecordBatchStream};
 
-/// `ExecutionPlan` represent nodes in the DataFusion Physical Plan.
+/// Represent nodes in the DataFusion Physical Plan.
 ///
-/// Each `ExecutionPlan` is partition-aware and is responsible for
-/// creating the actual `async` [`SendableRecordBatchStream`]s
-/// of [`RecordBatch`] that incrementally compute the operator's
-/// output from its input partition.
+/// Calling [`execute`] produces an `async` [`SendableRecordBatchStream`] of
+/// [`RecordBatch`] that incrementally computes a partition of the
```

Comment:
when we talk about partition we mean a separate task(thread) that processes a chunk of input data by applying `.execute()` method?

Reply:
I added a link to `Partitioning` in this PR and I have made another PR to try and document better what partitioning means: https://github.com/apache/arrow-datafusion/pull/8022 -- please take a look when you have time.